shap/_explanation.py

```diff
@@ -16,6 +17,15 @@
 op_chain_root = OpChain("shap.Explanation")
 
 
+@dataclass
+class OpHistoryItem:
```

Comment:
It might help to add a docstring to explain the purpose of this object, perhaps something like

```suggestion
class OpHistoryItem:
    """Represents a lazily-applied dot-chained operation, such as `.max()` or `.abs()`"""
```

Reply:
I think your description is somewhat correct, but I don't think it's too helpful to make a comparison between these two. `OpChain` is representing a set of dot-chained operations that can be applied to some object in the future via the `.apply(obj)` method.  
`OpChain._ops` stores the list of such ops, to be applied.

And `Explanation().op_history` stores the list of ops that have already been applied to the said Explanation object.

Technically, we could use `OpHistoryItem` to describe the ops that are in `OpChain._ops` as well, but we'll need to set the `.prev_shape` to be optional since it doesn't make sense to speak of a previous shape when the object on which to apply the ops on hasn't been supplied yet. I decided against this for now though, I'm not sure whether this is something we want to do -- they should be separate objects. That one could be an `OpChainItem` for example.

I'll add docstrings to this dataclass in the meantime, but I'll leave out the comparison.

Reply:
Thanks for the detailed explanation, that's helpful to help me build confidence in my understanding :)